src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py

```diff
@@ -32,22 +33,26 @@ def __init__(self, config):
         self.condorAPI = PyCondorAPI()
         self.agentConfig = {}
         self.validSpeedDrainConfigKeys = ['CondorPriority', 'NoJobRetries', 'EnableAllSites']
-
         self.reqAuxDB = ReqMgrAux(self.config.General.ReqMgr2ServiceURL)
+        self.emailAlert = EmailAlert(config)
 
     @timeFunction
     def algorithm(self, parameters):
         """
         Update drainStats if agent is in drain mode
         """
         logging.info("Running agent drain algorithm...")
+        # make a copy of the previous agent aux db configuration to compare against later
+        previousConfig = copy.deepcopy(self.agentConfig)
```

Comment:
Hmm, I see the following scenario:
cycle 1: component goes through the normal cycle
cycle 2: component fails to fetch the agent config from ReqMgrAux
cycle 3: an empty list gets assigned to previousConfig (since it failed to fetch the config in the previous cycle).

In order to properly handle such scenarion, we should then have something like:
```
if self.agentConfig:
    previousConfig = copy.deepcopy(self.agentConfig)
```

Have I missed anything?